README.md
# tk
Tekwill Final Project


changelog.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.1.1] - 2023-09-11

### Added
- Home-Work #13
- Home-Work #14
- Home-Work #15


script2.js
console.log("test");
// Fruits Array//
const fruits = [];
fruits.push("apple");
fruits.push("banana", "orange");
console.log("The lenght of fruits array is " + fruits.length);
console.log(fruits);

//Number array//

const numbers = [3, 4, 5];
numbers.pop();
console.log(numbers)
numbers.shift()
console.log(numbers)

// colors array//
const colors = ["red", "green", "blue", "yellow"]
console.log(colors.indexOf("green"))
console.log(colors.includes("orange"))

//names array//

const names = ["John", "Jane", "Mike", "Jenifer"]
names.unshift("David")
console.log(names)
console.log(names.indexOf("Mike"))

//months array//
const months = new Array("January", "February", "March", "April", "May", "June", "July", "August", "September",
    "October", "November", "December")
months.sort()
console.log(months)

const filteredArray = months.filter((month) => month.length > 3)

console.log(filteredArray)

const myMonths = ["January", "February", "March", "April", "May", "June", "July", "August", "September",
    "October", "November", "December"]

console.log(myMonths.slice(6, 11))

let fiveLetters = months.filter((month) => month.length > 5)
console.log(fiveLetters.join("-"))

script3.js
// Homework 17//
//Array Fruits//

const fruits = ["apple", "banana", "orange", "peach", "mango"]

for (let i = 0; i < fruits.length; i++) {
    console.log(fruits[i])
}

//Array Number//
let numbers = new Array(1, 2, 4, 5, 6)
let result = 0;
for (let i = 0; i < numbers.length; i++) {
    result += numbers[i]
}
console.log("The sum of numbers is array equals " + result)

//Array names//

const names = ["John", "JaneaJaneaJanea", "Michael", "Jennifer", "Bobby"]
const newArray = new Array()
for (let i = 0; i < names.length; i++) {
    newArray.push(names[i].length)
}
console.log(newArray)

//Department//

// const company = [
// ["Dev", "John", "Bob", "Samanta", "Peter"], 
// [ "QA", "Silivia", "Jane", "Monica", "Henry"],
// ["HR", "Megan", "Kate","Beth", "Eve"],
//  ["Finance", "Barbie", "Ken", "Marusia", "Lea"]
// ]

const company = [["Dev", "QA", "HR", "Finance"],
["John", "Bob", "Samanta", "Peter"],
["Silivia", "Jane", "Monica", "Henry"],
["Megan", "Kate", "Beth", "Eve"],
["Barbie", "Ken", "Marusia", "Lea"]

]
const sortedArray = new Array()
const myArray = new Array()
for (let i = 0; i < company.length; i++) {
    if (i > 0) {
        //  console.log(company[i])
        let name = company[i]
        sortedArray.push(company[i])
    }
}
console.log(sortedArray)
const finalArray = new Array
for (i = 0; i < sortedArray.length; i++) {
    for (j = 0; j < sortedArray[i].length; j++) {
        finalArray.push(sortedArray[i][j]);
    }
}
console.log(finalArray.sort())

// Objects //

const person = { fname: "John", name: "Doe", age: 30, job: "developer" }

function personPropreties(obj) {
    let i = 0
    for (let key in obj) { i++ }
}
console.log("Person characters " + i)

//Array random number//
function randomArray(n) {
    const ranArr = new Array()
    for (let i = 1; i <= n; i++) {
        //   console.log(i)
        ranArr.push(i)
    }
    console.log(ranArr)
}
randomArray(50)

//Biggest number//

const numbers2 = [5, 6, 20, 345, 80, 2, 3444, 40]
let theBiggestNumber = 0

for (let i = 0; i <= numbers2.length; i++) {
    if (numbers2[i] > theBiggestNumber) {
        theBiggestNumber = numbers2[i]
    }
}
console.log("The Biggest Number " + theBiggestNumber)

//Average //

let total = 0;
let count = 0;
function average(grades) {
    console.log(grades)

    for (let proprety in grades) {
        if (typeof grades[proprety] === 'number') {

            total += grades[proprety];

            count++;
        }

    }
    return total / count;
}
let grades = { math: 8, literature: 7, history: 9, biology: 5 }
console.log(average(grades));

// Object Dog (Pes)//

let dog = new Object();

function Dog(nickname, age, color) {
    this.name = nickname
    this.age = age
    this.color = color
    this.puppiesNames = [];
}

function addPuppies(dog, puppiesNumber) {
    for (let i = 1; i <= puppiesNumber; i++) {
        dog.puppiesNames.push("Puppy" + i);
    }
}

let myDog = new Dog("Buddy", 5, "Golden");

addPuppies(myDog, 3);

console.log(myDog.puppiesNames);
console.log(myDog)


script.js
// Home -work for lesson #13 //

let country = "Moldova";
let continent = "Europe";
let population = 2.615;

console.log("Country name is " + country);
console.log(country + "'s continent is " + continent);
console.log(country + "'s population is " + population + " billions");

let isIsland = false;

//let coutryDescription = country + " is located in " + continent + ", " + population + " mln people live here";

let countryDescription = `Country ${country} is located in ${continent}, ${population} mln people live here`;
console.log(countryDescription);



// Home -work for lesson #14 //

let weightSarah = 65;
let weightBob = 93;
let heightSarah = 1.75;
let heightBob = 1.75;

let sarahBMI = weightSarah / (Math.pow(heightSarah, 2));
let bobBMI = weightSarah / (Math.pow(heightBob, 2));
let bmiLow = 18.5;
let bmiMiddle = 25;
let bmiHigh = 25;
if (sarahBMI < bmiLow) {
    console.log("Sarah's BMI is " + sarahBMI + " underweighted");
} else {
    if (sarahBMI > bmiLow && sarahBMI < bmiMiddle) {
        console.log("Sarah's BMI is " + sarahBMI + " normal weight");
    } else {
        console.log("Sarah's BMI is " + sarahBMI + " overwiehgt");

    }
}
switch (bobBMI) {
    case 0:
        bobBMI < bmiLow;
        console.log("Bob's BMI is " + bobBMI + " underweighted");
        break;
    case 1:
        bobBMI > bmiLow && bobBMI < bmiMiddle;
        console.log("Bob's BMI is " + bobBMI + " normal weight");
        break;
    default:
        console.log("Bob's BMI is " + bobBMI + " overweight");

}

// Home -work for lesson #15 //

const NUMBER = 42;
function isOddEven() {
    let result = NUMBER / 2
    // console.log(result);
    if (result === 0) {
        console.log(NUMBER + " is Even")
    } else
        console.log(NUMBER + " is Odd")
}
isOddEven();

// Ternary operator//


let isLoggedIn = true;

let result = (isLoggedIn) ? 'logged in' : 'logged out';
console.log("You're " + result);

if (isLoggedIn) {
    console.log("Hello Username!");
} else {
    console.log("Please log in");
}

// Warehouse items//

let itemPrice = 100;
let quantaty = 3000000;
let isInStock = true;
//let isInStock = false;

let itemsValue = isInStock * quantaty

if (isInStock) {
    console.log("The value of good on deposit is " + itemsValue + "$")
} else {
    console.log("There are not items in stock")
}

//  Leap Year//

const currentDate = new Date();
const year = currentDate.getFullYear();


let resultYear = year % 4;
if (resultYear == 0) {
    console.log("The year " + year + " is leap year");
} else {
    console.log("The year " + year + " is not a leap year");
}

// Switch //

let countryCode = "US";
switch (countryCode) {
    case "MD":
        console.log("Moldova");
        break;
    case "US":
        console.log("United States");
        break;

    case "UK":
        console.log("Great Britain");
        break;

    case "ZM":
        console.log("Zimbabve");
        break;

    case "SO":
        console.log("Somali");
        break;

    case "RO":
        console.log("Romania");
        break;
    default:
        console.log("We don't know that country yet.");

}
const grade = 87;

//
switch (true) {
    // If score is 90 or greater
    case grade >= 90:
        console.log("A");
        break;
    // If score is 80 or greater
    case grade >= 80:
        console.log("B");
        break;
    // If score is 70 or greater
    case grade >= 70:
        console.log("C");
        break;
    // If score is 60 or greater
    case grade >= 60:
        console.log("D");
        break;
    // Anything 59 or below is failing
    default:
        console.log("F");
}
// Functions //

function TellFortune(n, z, y, x) {
    let tellFortune = `You will be ${x} in ${y}, and get married to ${z} with ${n} kids`;
    console.log(tellFortune);
    //console.log("You will be " + x + " in " + y + " , and get married to " + z + " with " + n + " kids")
}
TellFortune(2, "Eve", "Heaven", "CTO")
TellFortune(0, "John", "Thailand", "CEO")
TellFortune((Math.floor((Math.random() * 4) + 1)), "Sarah", "USA", "BA")

//DOG//

function CalculateDogAge(dogAge, humanAge) {
    let dogAgeResult = dogAge * 7;
    console.log("Your dog is " + dogAgeResult + " in dog's years");
    let humanAgeResult = humanAge / 7;
    console.log("HumanAge " + humanAgeResult)

}


CalculateDogAge(3, 70)
CalculateDogAge(4)
CalculateDogAge(5, 40)
CalculateDogAge(7);


// Arrow Functions //

const func = (curAge, food, maxYears) => (maxYears - curAge) * (food * 365);
func(33, 3, 100)

console.log(func(33, 3, 100))
console.log(func(15, 2, 100))
console.log(func(40, 1, 100))


